app/utilities/mappings.py

```diff
@@ -35,3 +35,8 @@ def _handle_sequence(value: Sequence) -> Generator[Mapping, None, None]:
                 yield from get_mappings_with_key(key, v, ignore_keys)
             if isinstance(v, Sequence):
                 yield from _handle_sequence(v)
+
+
+def get_values_for_key(key: str, data: Mapping | Sequence) -> Generator:
```

Comment:
Minor, I suggest we enforce keyword only arg at least for data.

Reply:
Sure, whats your general view for enforcing? always do it for 2+? Is there somewhere we have/can add a general rule for this? 

The approach I have been taking is whenever there is more than one argument, enforce kwargs, unless theres only a couple of argument types and the types don't overlap, such as here, because mypy would pick that up. (Or exceptional cases like SectionKey where its very common and enforcing would bloat) but I'm happy to go with enforcing anytime its more than 1

Reply:
We don't have a set rule (yet). My preference is when the argument order becomes unclear. I wasn't suggesting enforcing keyword args for `key` given the function name end with `key`, so I was thinking `get_values_for_key(key: str, *, data: Mapping | Sequence) -> ....`. This one was possibly fine due to function name, i.e name implies first should be key, then there is only 1 other.

We can come up with our preference for when we create a style guide. I agree 2+ is a good rule, or if function name doesn't make clear the arg order for the first two.

> because mypy would pick that up
Mypy won't pick up bad test cases in which the order is different as we don't enforce mypy for tests.

Reply:
Have taken out the enforcing of key, I think that makes sense, good with 2+/ unclear